### DISCUSSION TITLE
Is there a way to use the resource dictionary in the action session?

### DISCUSSION THREAD

**Discussion by yufuluo** [cloud-custodian/cloud-custodian#7474] [AWS]
Is there a way to query the resource dictionary and use the result as a param in the action session? 
For example, I would like to enable bucket key for buckets using KMS keys in SSE. But I don't want to change the current KMS key id for each bucket. Is there a way to query the resource dict and use the result string? 
something like the following:
```policies:
  - name: s3-enable-kms-bucket-key
    resource: s3
    filters:
      - type: bucket-encryption
        state: true
        crypto: aws:kms
        bucket-key: false
    comment: |
      Enable bucket key for KMS encryption
    actions:
      - type: set-bucket-encryption
        crypto: aws:kms
        key: "resource.c7n:bucket-encryption.Rules.ServerSideEncryptionConfiguration[0].ApplyServerSideEncryptionByDefault.KMSMasterKeyID"
        bucket-key: true

**Comment by thisisshi** [comment]
currently no, this would require an enhancement to the action or potentially a separate action but that does make sense.
